lib/code_corps_web/plugs/data_to_attributes.ex

```diff
@@ -12,13 +12,34 @@ defmodule CodeCorpsWeb.Plug.DataToAttributes do
   def init(opts), do: opts
 
   @spec call(Conn.t, Keyword.t) :: Plug.Conn.t
-  def call(%Conn{params: %{"data" => data} = params} = conn, _opts) do
+  def call(%Conn{params: %{} = params} = conn, many_mappings) do
     attributes =
       params
       |> Map.delete("data")
-      |> Map.merge(data |> JaSerializer.Params.to_attributes)
+      |> Map.delete("included")
+      |> Map.merge(params |> parse_data)
+      |> Map.merge(params |> parse_included(many_mappings))
 
     conn |> Map.put(:params, attributes)
   end
   def call(%Conn{} = conn, _opts), do: conn
+
+  defp parse_data(%{"data" => data}), do: data |> JaSerializer.Params.to_attributes
+  defp parse_data(%{}), do: %{}
+
+  defp parse_included(%{"included" => included}, many_mappings) do
+    included |> Enum.reduce(%{}, fn (%{"data" => %{"type" => type}} = params, parsed) ->
+      attributes = params |> parse_data()
+
+      case many_mappings |> Map.has_key?(type) do
+        true ->
+          parsed
+          |> Map.update(many_mappings[type], attributes, fn (records, record) ->
+            records ++ [record]
+          end)
+        false -> parsed |> Map.put(type, attributes)
+      end
```

Comment:
Is there any reason to do a `case` here over an `if/else`?

Reply:
I had to fix this fn because `Map.update`'s fun parameter only takes a single value (in this case the accumulated records) and we needed to provide the `cast_assoc` with an array with the default argument if the key doesn't yet exist (i.e. `[attributes]`).

I _think_ this is the logic you were looking for @begedin. Spent quite some time debugging and trying to figure this out. More than I'm happy to admit.